### DISCUSSION TITLE
Need to calculate moonrise time based on a date and location

### DISCUSSION THREAD

**Discussion by SaqibZafar** [DeepskyLog/laravel-astronomy-library#86] [Q&A] [answered]
Hi guys, i am trying to get moonrise time using the $moon object, which function will give me the moon data, i already have the following code working great.

        $geo_coords = new GeographicalCoordinates($longitude,$latitude);
        // Calculate the Equatorial Coordinates of the Moon
        $moondate   = Carbon::create($year, $month, $day, $hour);
        $moon  = new Moon();
        $moon->calculateApparentEquatorialCoordinates($moondate);
        // If the correction for the location on earth should be done, use the following method
        $heightOfLocation = 10;   // In meters
        $moon->calculateEquatorialCoordinates($moondate, $geo_coords, $heightOfLocation);

        $RA = round($moon->getEquatorialCoordinates()->getRA()->getCoordinate(),2);
        $DEC = round($moon->getEquatorialCoordinates()->getDeclination()->getCoordinate(),2);
        
        $moonRiseTime = ?

**Comment by WimDeMeester** [comment, marked as answer]
Does something like this work?

```php
$astrolib = \App\Models\Astrolib::getInstance()->getAstronomyLibrary();

$geo_coords = new GeographicalCoordinates($longitude,$latitude);
// Calculate the Equatorial Coordinates of the Moon
$moondate   = Carbon::create($year, $month, $day, $hour);
$moon  = new Moon();
   
$greenwichSiderialTime = deepskylog\AstronomyLibrary\Time::apparentSiderialTimeGreenwich($moondate->subDay());
$deltaT = $astrolib->getDeltaT();

$moon = new \deepskylog\AstronomyLibrary\Targets\Moon();
$heightOfLocation = 10;   // In meters
$moon->calculateEquatorialCoordinates($moondate, $geo_coords, $heightOfLocation);
$moon->calculateEphemerides($geo_coords, $greenwichSiderialTime, $deltaT);
// getTransit and getSetting also work
$moon->getRising()->timezone($location->timezone)->format('H:i');



```
